gh-update-branch

```diff
@@ -1,8 +1,22 @@
 #!/bin/bash
 set -e
 
-read -r number sha < <(gh pr view ${1} --json number,commits --jq '[.number,.commits[-1].oid]|@tsv')
+if [ -z "$1" ]; then
+  prs=$(gh pr view --jq '.[].number' --json number)
+  echo "Updating current PR"
+elif [ $1 == "--all" ]; then
+  prs=$(gh pr list --author @me --jq '.[].number' --json number)
+  echo "Updating all your PRs"
+else
+  prs=($1)
+  echo "Updating specified PR"
+fi
 
-exec gh api --silent "repos/{owner}/{repo}/pulls/${number}/update-branch" \
-  --field expected_head_sha=${sha} \
-  --method PUT
+for pr in ${prs[@]}; do
+  read -r number sha < <(gh pr view $pr --json number,commits --jq '[.number,.commits[-1].oid]|@tsv')
+  echo "Updating PR #${number}"
```

Comment:
I believe this results in duplicate output in the single case, right?

Reply:
The reason is that, in general, one echo tells what option has been parsed from the first argument and the second echo tells which PRs are being updated